drivers/gpio/gpio_mchp_xec.c

```diff
@@ -66,18 +77,12 @@ static int gpio_xec_configure(const struct device *dev,
 
 	/* The flags contain options that require touching registers in the
 	 * PCRs for a given GPIO. There are no GPIO modules in Microchip SOCs!
-	 *
-	 * Start with the GPIO module and set up the pin direction register.
-	 * 0 - pin is input, 1 - pin is output
+	 * Keep direction as input until last.
+	 * Clear input pad disable allowing input pad to operate.
 	 */
 	mask |= MCHP_GPIO_CTRL_DIR_MASK;
 	mask |= MCHP_GPIO_CTRL_INPAD_DIS_MASK;
-	if ((flags & GPIO_OUTPUT) != 0U) {
-		pcr1 |= MCHP_GPIO_CTRL_DIR_OUTPUT;
-	} else {
-		/* GPIO_INPUT */
-		pcr1 |= MCHP_GPIO_CTRL_DIR_INPUT;
-	}
+	pcr1 |= MCHP_GPIO_CTRL_DIR_INPUT;
```

Comment:
@scottwcpg Looks ok. I have a question is this going to allow us to reconfigure without glitches. Let's say we power a board and then configure it for the first time as input, then we reconfigure it as output later in the flow. I want to understand if this is going to be possible.  I don't know if you want to keep line 85 maybe just for documenting the flow with code ?

Reply:
Line 85 is necessary to allow preloading the desired output value before setting the direction to output. I ran quicks tests looping where I forced the pin control register to its POR state 0x8040 and called the API to setting it to output high. The pin had an external pull-up and a scope attached. I observed no glitching but this is not an exhausted test of every possible initial state of the pin control register and API call (flags, state).
If the pin was configured previously as an input and code requests it become an output the new code keeps the pin an input until the desired output value is loaded and then switches direction to output. Depending on external HW and internal pulls the desired output state can cause a change in the pin as expected. I believe the case you are concerned about is the pin has been configured as an output or the pin's POR state is not 0x8040 but actually configured as an output. This could cause a glitch. Logic needs to be added to this routine to check the pin's direction setting before making changes. If the pin is already an output then there's no need to switch to input. The new code will have to look at both direction and AOD before deciding how to change the output state.  I will start work on the change.

Reply:
Thanks for the explanation @scottwcpg